tools/reviewer/job/impl/ReviewerMetadataUpdaterTask.java

```diff
@@ -0,0 +1,182 @@
+    this.fileUtils = fileUtils;
+    this.gitRepoFactory = gitRepoFactory;
+    this.firestoreClientFactory = firestoreClientFactory;
+  }
+
+  private void uploadProtoToFirestore(ReviewerRegistry registry) {
+    firestoreClient.createDocument(
+        REVIEWER_REGISTRY_COLLECTION, REVIEWER_REGISTRY_DOCUMENT_NAME, registry);
+    firestoreClient.createProtoDocument(
+        REVIEWER_REGISTRY_COLLECTION, REVIEWER_REGISTRY_DOCUMENT_NAME_BIN, registry);
+  }
+
+  private void initializeFirestoreClientIfNull() {
+    if (this.firestoreClient == null) {
+      FileInputStream serviceAccount = null;
+      try {
+        serviceAccount = new FileInputStream(Job.serviceAccountJson.get());
+        ServiceAccountCredentials cred =
+            (ServiceAccountCredentials)
+                GoogleCredentials.fromStream(serviceAccount)
+                    .createScoped(
+                        Arrays.asList(
+                            "https://www.googleapis.com/auth/cloud-platform",
+                            "https://www.googleapis.com/auth/datastore"));
+        this.firestoreClient =
+            this.firestoreClientFactory.create(
+                cred.getProjectId(), cred.refreshAccessToken().getTokenValue());
+      } catch (IOException e) {
+        e.printStackTrace();
+      }
+    }
+  }
+
+  private static Stream<Integer> intStream(byte[] array) {
+    return IntStream.range(0, array.length).map(idx -> array[idx]).boxed();
+  }
+
+  private static String md5ForFile(String filePath) throws IOException {
+    byte[] fileAsBytes = Files.readAllBytes(Paths.get(filePath));
+    try {
+      MessageDigest messageDigest = MessageDigest.getInstance("MD5");
+      messageDigest.update(fileAsBytes);
+      byte[] messageDigestBytes = messageDigest.digest();
+      StringBuilder hashBuilder = new StringBuilder();
+      intStream(messageDigestBytes)
+          .map(digestByte -> String.format("%02X", Byte.toUnsignedInt(digestByte.byteValue())))
+          .forEach(hashBuilder::append);
+      return hashBuilder.toString();
+    } catch (NoSuchAlgorithmException ex) {
+      // fallback to full file contents
+      // this should not happen, MD5 is widely available
+      return new String(fileAsBytes);
+    }
+  }
+
+  @Override
+  public void run() {
+    if (lock.tryLock()) {
+      try {
+        String globalRegistryFilePath =
+            fileUtils.joinToAbsolutePath(REPO_DIRECTORY, GLOBAL_REGISTRY_FILE);
+        initializeFirestoreClientIfNull();
+        if (storedChecksum == null) {
+
+          GitRepo repo = gitRepoFactory.create(REPO_DIRECTORY);
+
+          if (fileUtils.folderEmptyOrNotExists(REPO_DIRECTORY)) {
+            repo.cloneRepo(Job.repoUrl.get(), REPO_DIRECTORY);
+          } else {
+            repo.pull();
+          }
+
+          storedChecksum = md5ForFile(globalRegistryFilePath);
+          log.atInfo().log("Computed checksum: %s", storedChecksum);
+
+          ReviewerRegistry reg =
```

Comment:
I think we can simplify this function by merging the main if-else into a single flow:
* Make sure file is fresh, pull/clone as needed
* Get checksum
* If not equal to stored, whether because stored is null or not
* Do the main action and update stored